app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -55,10 +67,21 @@ class SearchRepositoriesViewModel(private val repository: GithubRepository) : Vi
         if (visibleItemCount + lastVisibleItemPosition + VISIBLE_THRESHOLD >= totalItemCount) {
             val immutableQuery = queryLiveData.value
             if (immutableQuery != null) {
+                _repoLoadStatus.postValue(LoadState.Loading)
                 viewModelScope.launch {
                     repository.requestMore(immutableQuery)
                 }
             }
         }
     }
-}
\ No newline at end of file
+
+    fun retry() {
+        val immutableQuery = queryLiveData.value
+        if (immutableQuery != null) {
```

Comment:
You can avoid the local val here with `queryLiveData.value?.let { immutableQuery -> .. }` but I'm not sure if that's better..